sensors/accelerometer/accelerometer.ino

```diff
@@ -0,0 +1,25 @@
+#include "accelerometer.h"
+#include <math.h>
+#include <SPI.h>
+// https://github.com/sparkfun/SparkFun_LSM9DS1_Arduino_Library/archive/master.zip
+#include <SparkFunLSM9DS1.h>
+
+LSM9DS1 accelerometer;
+
+Accelerometer::Accelerometer(const int &pin) : Sensor(&pin) {}
+
+void Accelerometer::read_sensor_value() {
+  // with no arguments, this uses default addresses (AG:0x6B, M:0x1E) and i2c port (Wire).
+  if (accelerometer.begin() == false) {
+    Serial.println("Failed to communicate with LSM9DS1.");
+  } else if (accelerometer.accelAvailable()) {
+    accelerometer.readAccel();
+  }
+  
+  // Now we can use the ax, ay, and az variables as we please.
+  // Either use them as raw ADC values, or calculated in g's.
+  raw_value = sqrt(pow(accelerometer.ax, 2) + pow(accelerometer.ay, 2) + pow(accelerometer.az, 2));
+  m_raw_value = raw_value;
+  sensor_value = sqrt(pow(accelerometer.calcAccel(accelerometer.ax), 2) + pow(accelerometer.calcAccel(accelerometer.ay), 2) + pow(accelerometer.calcAccel(accelerometer.az), 2));
```

Comment:
Hmm I'm a little concerned that we're losing data here amidst the processing. I'm happy keeping it like this though if we know that we'll never want to use the individual components of the acceleration. 

Generally, this points to a potential shortcoming of our current base sensor implementation, which can store consumable data as a single `double`. What about something like a GPS module that gives us position in 3 different coordinates? Maybe we want a different data structure instead of `double` to hold that in, idk. Thoughts @ashvinnagarajan?

Reply:
Me and Ash actually discussed it yesterday, and we decided that I would develop a "cartesian_sensor" class derived from "base_sensor" that would include xyz data members. The sensors that currently fall under its scope are the accelerometer, gyroscope, and magnetometer, so it makes sense to develop a single base class for them.

Reply:
For compute_magnitude() function, will be implemented in the base class as it is a common function to all the sensors.